unittests/AST/ASTImporterTest.cpp

```diff
@@ -3173,6 +3173,27 @@ TEST_P(
                     .match(ToTU, classTemplateSpecializationDecl()));
 }
 
+TEST_P(ASTImporterTestBase, MergeOfRecordAtImport) {
```

Comment:
Isn't this an ODR violation? Two TU with different definitions?

Reply:
This can be ODR problem but it is not checked yet. It was also not checked before this fix. (To check this the structurally equivalence check must check the methods of a record.)

In an encountered error case there is an existing `CXXRecordDecl` that do not contain destructor, and another one is imported that does contain an automatically generated destructor function. The two classes are in separate header files but otherwise exactly the same source code. It is another question why the original record does not have the destructor but I did not try to figure this out, I do not know at all where to look for related code.

The test is there only to check if the methods are merged. A normal use case is when the first class has no generated destructor (or operators), the second (imported) has these functions, and the source code is the same.

Reply:
OK.
Only the automatically generated functions were missing? Or was there any other user defined functions which was missing?